java/server/src/org/openqa/selenium/grid/log/LoggingOptions.java

```diff
@@ -102,7 +113,63 @@ private Tracer createTracer() {
     exporters.add(SimpleSpanProcessor.newBuilder(new SpanExporter() {
       @Override
       public ResultCode export(Collection<SpanData> spans) {
-        spans.forEach(span -> LOG.fine(String.valueOf(span)));
+
+        spans.forEach(span -> {
+          LOG.fine(String.valueOf(span));
+
+          String traceId = span.getTraceId().toLowerBase16();
+          String spanId = span.getSpanId().toLowerBase16();
+          Status status = span.getStatus();
+          List<Event> eventList = span.getEvents();
+          eventList.forEach(event -> {
+            Map<String, Object> map = new HashMap<>();
+            map.put("eventTime", event.getEpochNanos());
+            map.put("traceId", traceId);
+            map.put("spanId", spanId);
+            map.put("spanKind", span.getKind().toString());
+            map.put("eventName", event.getName());
+
+            Attributes attributes = event.getAttributes();
+            Map<String, Object> attributeMap = new HashMap<>();
+            attributes.forEach((key, value) -> {
+              Object attributeValue = null;
+              switch (value.getType()) {
+                case LONG:
+                  attributeValue = value.getLongValue();
+                  break;
+                case DOUBLE:
+                  attributeValue = value.getDoubleValue();
+                  break;
+                case STRING:
+                  attributeValue = value.getStringValue();
+                  break;
+                case BOOLEAN:
+                  attributeValue = value.getBooleanValue();
+                  break;
+                case STRING_ARRAY:
+                  attributeValue = value.getStringArrayValue();
+                  break;
+                case LONG_ARRAY:
+                  attributeValue = value.getLongArrayValue();
+                  break;
+                case DOUBLE_ARRAY:
+                  attributeValue = value.getDoubleArrayValue();
+                  break;
+                case BOOLEAN_ARRAY:
+                  attributeValue = value.getBooleanArrayValue();
```

Comment:
This case statement makes me twitch. It's long, and it looks like a prime candidate to accidentally forget to extend if we add a new value type. At the least, I'd suggest making `Attributes` be aware of how to serialise itself to JSON, so you can do a `map.put("attributes", event.getAttributes)`

Reply:
I understand the concern expressed. However, the OpenTelemetry `Attributes` class stores a `String` key and `AttributeValue` value and in order to get the actual value from `AttributeValue`, it offers series of methods of based on the `AttributeValue.Type`. So this is the only way I am seeing to creating the attributeMap from Events.

Reply:
Initially I considered extending either the Event or Attributes to add a method for Json serialisation of the attributes. After looking into the OTel classes, the `Attributes` implementation `AutoValue_Attributes_ArrayBackedAttributes` is final and cannot be extended. Same is applicable for `Events` implementation `AutoValue_TimedEvent_RawTimedEvent`. So we might be stuck with the switch case. Any suggestions on this front?